diff --git a/crypto/asn1/a_d2i_fp.c b/crypto/asn1/a_d2i_fp.c
index f152f348aa..d537f88a94 100644
--- a/crypto/asn1/a_d2i_fp.c
+++ b/crypto/asn1/a_d2i_fp.c
@@ -187,6 +187,7 @@ int asn1_d2i_read_bio(BIO *in, BUF_MEM **pb)
         } else {
             /* suck in slen bytes of data */
             want = slen;
+IJON_MAX((unsigned long long)slen); /* PATCHID:44149 */
             if (want > (len - off)) {
                 want -= (len - off);
 #ifdef MAGMA_ENABLE_FIXES
@@ -200,6 +201,7 @@ int asn1_d2i_read_bio(BIO *in, BUF_MEM **pb)
                     goto err;
                 }
                 while (want > 0) {
+IJON_MAX((unsigned long long)want); /* PATCHID:43692 */
                     /*
                      * Read content in chunks of increasing size
                      * so we can return an error for EOF without
@@ -207,6 +209,7 @@ int asn1_d2i_read_bio(BIO *in, BUF_MEM **pb)
                      * in one go.
                      */
                     size_t chunk = want > chunk_max ? chunk_max : want;
+IJON_SET((int)chunk); /* PATCHID:42723 */
 
                     if (!BUF_MEM_grow_clean(b, len + chunk)) {
                         ERR_raise(ERR_LIB_ASN1, ERR_R_MALLOC_FAILURE);
@@ -215,6 +218,8 @@ int asn1_d2i_read_bio(BIO *in, BUF_MEM **pb)
                     want -= chunk;
                     while (chunk > 0) {
                         i = BIO_read(in, &(b->data[len]), chunk);
+IJON_CMP((unsigned long long)i, 0); /* PATCHID:42159 */
+IJON_MIN((unsigned long long)i); /* PATCHID:41630 */
 #ifdef MAGMA_ENABLE_CANARIES
                         /**
                          * An EOF is encountered before the requested/allocated
diff --git a/crypto/asn1/a_int.c b/crypto/asn1/a_int.c
index bfab1bb2c2..771119561a 100644
--- a/crypto/asn1/a_int.c
+++ b/crypto/asn1/a_int.c
@@ -147,10 +147,12 @@ static size_t i2c_ibuf(const unsigned char *b, size_t blen, int neg,
     else if (!neg)
         memcpy(p, b, (unsigned int)blen);
     else {
+IJON_CMP((unsigned long long)neg, 1); /* PATCHID:45101 */
         /* Begin at the end of the encoding */
         const unsigned char *n = b + blen - 1;
         p += blen - 1;
         i = blen;
+IJON_SET((int)blen); /* PATCHID:44734 */
 
         /**
          * The representation of a negative zero is just all zeros. This routine
@@ -162,9 +164,12 @@ static size_t i2c_ibuf(const unsigned char *b, size_t blen, int neg,
 
         /* Copy zeros to destination as long as source is zero */
         while (!*n) {
+IJON_CMP((unsigned long long)*n, 0); /* PATCHID:44612 */
             *(p--) = 0;
             n--;
             i--;
+IJON_SET((int)i); /* PATCHID:44579 */
+IJON_MIN((unsigned long long)i); /* PATCHID:44467 */
 #ifdef MAGMA_ENABLE_CANARIES
             /**
              * The value of `i` determines how many more iterations this loop
diff --git a/crypto/asn1/a_type.c b/crypto/asn1/a_type.c
index 280e09883c..06b213f138 100644
--- a/crypto/asn1/a_type.c
+++ b/crypto/asn1/a_type.c
@@ -105,6 +105,7 @@ int ASN1_TYPE_cmp(const ASN1_TYPE *a, const ASN1_TYPE *b)
     case V_ASN1_OTHER:
     default:
 #ifdef MAGMA_ENABLE_CANARIES
+IJON_CMP((unsigned long long)a->type, (unsigned long long)V_ASN1_BOOLEAN); /* PATCHID:46026 */
     MAGMA_LOG("%MAGMA_BUG%", a->type == V_ASN1_BOOLEAN);
 #endif
         result = ASN1_STRING_cmp((ASN1_STRING *)a->value.ptr,
diff --git a/crypto/bn/bn_gf2m.c b/crypto/bn/bn_gf2m.c
index 25673d6d24..4cbffb6f39 100644
--- a/crypto/bn/bn_gf2m.c
+++ b/crypto/bn/bn_gf2m.c
@@ -650,6 +650,7 @@ static int BN_GF2m_mod_inv_vartime(BIGNUM *r, const BIGNUM *a,
                                  * aggressive. But we don't have to "cache"
                                  * p->d, because *p is declared 'const'... */
         while (1) {
+IJON_INC((int)ubits); /* PATCHID:22488 */
             while (ubits && !(udp[0] & 1)) {
                 BN_ULONG u0, u1, b0, b1, mask;
 
@@ -669,8 +670,10 @@ static int BN_GF2m_mod_inv_vartime(BIGNUM *r, const BIGNUM *a,
                 bdp[i] = b0 >> 1;
                 ubits--;
             }
+IJON_MIN((unsigned long long)ubits); /* PATCHID:21645 */
 
 #ifdef MAGMA_ENABLE_FIXES
+IJON_CMP((unsigned long long)ubits, (unsigned long long)BN_BITS2); /* PATCHID:21327 */
             if (ubits <= BN_BITS2) {
                 if (udp[0] == 0) /* poly was reducible */
                     goto err;
@@ -680,6 +683,7 @@ static int BN_GF2m_mod_inv_vartime(BIGNUM *r, const BIGNUM *a,
 #else
 #ifdef MAGMA_ENABLE_CANARIES
             MAGMA_LOG("%MAGMA_BUG%", udp[0] == 0);
+IJON_CMP((unsigned long long)udp[0], (unsigned long long)0); /* PATCHID:20373 */
 #endif
             if (ubits <= BN_BITS2 && udp[0] == 1)
                     break;
@@ -713,6 +717,7 @@ static int BN_GF2m_mod_inv_vartime(BIGNUM *r, const BIGNUM *a,
                     utop--;
                 ubits = utop * BN_BITS2 + BN_num_bits_word(ul);
             }
+IJON_SET((int)ubits); /* PATCHID:19517 */
         }
         bn_correct_top(b);
     }
diff --git a/crypto/bn/bn_local.h b/crypto/bn/bn_local.h
index d9cdecc110..c1087b0ba5 100644
--- a/crypto/bn/bn_local.h
+++ b/crypto/bn/bn_local.h
@@ -663,6 +663,8 @@ BIGNUM *int_bn_mod_inverse(BIGNUM *in,
 
 static ossl_inline BIGNUM *bn_expand(BIGNUM *a, int bits)
 {
+IJON_DIST((long long) bits, (long long) (INT_MAX - BN_BITS2 + 1)); /* PATCHID:16706 */
+IJON_CMP((unsigned long long) bits, (unsigned long long) (INT_MAX - BN_BITS2 + 1)); /* PATCHID:16183 */
 #ifdef MAGMA_ENABLE_FIXES
     if (bits > (INT_MAX - BN_BITS2 + 1))
         return NULL;
diff --git a/crypto/evp/encode.c b/crypto/evp/encode.c
index b3f2dca109..27a47fd3e0 100644
--- a/crypto/evp/encode.c
+++ b/crypto/evp/encode.c
@@ -169,6 +169,9 @@ int EVP_EncodeUpdate(EVP_ENCODE_CTX *ctx, unsigned char *out, int *outl,
     if (inl <= 0)
         return 0;
     OPENSSL_assert(ctx->length <= (int)sizeof(ctx->enc_data));
+IJON_MAX((unsigned long long)inl); /* PATCHID:40973 */
+IJON_CMP((unsigned long long)inl, (unsigned long long)1000000); /* PATCHID:40607 */
+IJON_MAX((unsigned long long)ctx->num); /* PATCHID:39905 */
 #ifdef MAGMA_ENABLE_FIXES
     if (ctx->length - ctx->num > inl) {
 #else
@@ -178,8 +181,11 @@ int EVP_EncodeUpdate(EVP_ENCODE_CTX *ctx, unsigned char *out, int *outl,
     * of input data
     */
     MAGMA_LOG("%MAGMA_BUG%", INT_MAX - inl < ctx->num);
+IJON_CMP((unsigned long long)(INT_MAX - inl), (unsigned long long)ctx->num); /* PATCHID:39863 */
+IJON_DIST((long long)(ctx->num + inl), (long long)INT_MAX); /* PATCHID:39862 */
 #endif
     if ((ctx->num + inl) < ctx->length) {
+IJON_INC((int)((ctx->num + inl) < ctx->length)); /* PATCHID:39006 */
 #endif
         memcpy(&(ctx->enc_data[ctx->num]), in, inl);
         ctx->num += inl;
diff --git a/crypto/mdc2/mdc2dgst.c b/crypto/mdc2/mdc2dgst.c
index 339be4adc9..0c4a2c1c82 100644
--- a/crypto/mdc2/mdc2dgst.c
+++ b/crypto/mdc2/mdc2dgst.c
@@ -47,6 +47,8 @@ int MDC2_Update(MDC2_CTX *c, const unsigned char *in, size_t len)
     size_t i, j;
 
     i = c->num;
+IJON_MAX((unsigned long long)len); /* PATCHID:47944 */
+IJON_SET((int)i); /* PATCHID:47076 */
     if (i != 0) {
 #ifdef MAGMA_ENABLE_FIXES
         if (len < MDC2_BLOCK - i) {
@@ -54,6 +56,7 @@ int MDC2_Update(MDC2_CTX *c, const unsigned char *in, size_t len)
 #ifdef MAGMA_ENABLE_CANARIES
         MAGMA_LOG("%MAGMA_BUG%", SIZE_MAX - len < i);
 #endif
+IJON_CMP((unsigned long long)len, (unsigned long long)MDC2_BLOCK); /* PATCHID:46751 */
         if (i + len < MDC2_BLOCK) {
 #endif
             /* partial block */
diff --git a/crypto/pkcs7/pk7_doit.c b/crypto/pkcs7/pk7_doit.c
index b51a678a98..e9d1dc8183 100644
--- a/crypto/pkcs7/pk7_doit.c
+++ b/crypto/pkcs7/pk7_doit.c
@@ -243,11 +243,20 @@ BIO *PKCS7_dataInit(PKCS7 *p7, BIO *bio)
 #else
 #ifdef MAGMA_ENABLE_CANARIES
     MAGMA_LOG("%MAGMA_BUG%", MAGMA_OR(p7 == NULL, p7->d.ptr));
+IJON_SET((int)(p7 == NULL)); /* PATCHID:38353 */
+IJON_CMP((unsigned long long)p7, 0); /* PATCHID:38078 */
 #endif
 #endif
 
     i = OBJ_obj2nid(p7->type);
+IJON_CMP((unsigned long long)i, 22); /* PATCHID:37827 */
+IJON_CMP((unsigned long long)i, 25); /* PATCHID:37076 */
+IJON_CMP((unsigned long long)i, 21); /* PATCHID:36716 */
+IJON_CMP((unsigned long long)i, 23); /* PATCHID:36154 */
+IJON_CMP((unsigned long long)i, 113); /* PATCHID:35132 */
+IJON_SET((int)i); /* PATCHID:34435 */
     p7->state = PKCS7_S_HEADER;
+IJON_INC((int)i); /* PATCHID:33825 */
 
     switch (i) {
     case NID_pkcs7_signed:
@@ -437,6 +446,10 @@ BIO *PKCS7_dataDecode(PKCS7 *p7, EVP_PKEY *pkey, BIO *in_bio, X509 *pcert)
 #endif
 #endif
     i = OBJ_obj2nid(p7->type);
+IJON_CMP((unsigned long long)i, (unsigned long long)NID_pkcs7_enveloped); /* PATCHID:33342 */
+IJON_CMP((unsigned long long)i, (unsigned long long)NID_pkcs7_signedAndEnveloped); /* PATCHID:32482 */
+IJON_CMP((unsigned long long)i, (unsigned long long)NID_pkcs7_signed); /* PATCHID:31691 */
+IJON_SET((int)i); /* PATCHID:29012 */
     p7->state = PKCS7_S_HEADER;
 
     switch (i) {
@@ -448,6 +461,7 @@ BIO *PKCS7_dataDecode(PKCS7 *p7, EVP_PKEY *pkey, BIO *in_bio, X509 *pcert)
          * or if the contentType is wrong (i.e., not "data").
          */
         data_body = PKCS7_get_octet_string(p7->d.sign->contents);
+IJON_CMP((unsigned long long)(data_body == NULL ? 1 : 0), (unsigned long long)1); /* PATCHID:28236 */
         if (!PKCS7_is_detached(p7) && data_body == NULL) {
             ERR_raise(ERR_LIB_PKCS7, PKCS7_R_INVALID_SIGNED_DATA_TYPE);
             goto err;
@@ -457,8 +471,10 @@ BIO *PKCS7_dataDecode(PKCS7 *p7, EVP_PKEY *pkey, BIO *in_bio, X509 *pcert)
     case NID_pkcs7_signedAndEnveloped:
         rsk = p7->d.signed_and_enveloped->recipientinfo;
         md_sk = p7->d.signed_and_enveloped->md_algs;
+IJON_SET((int)(data_body == NULL)); /* PATCHID:30799 */
         /* data_body is NULL if the optional EncryptedContent is missing. */
         data_body = p7->d.signed_and_enveloped->enc_data->enc_data;
+IJON_CMP((unsigned long long)(data_body == NULL ? 1 : 0), (unsigned long long)1); /* PATCHID:27597 */
         enc_alg = p7->d.signed_and_enveloped->enc_data->algorithm;
 
         OBJ_obj2txt(name, sizeof(name), enc_alg->algorithm, 0);
@@ -479,9 +495,11 @@ BIO *PKCS7_dataDecode(PKCS7 *p7, EVP_PKEY *pkey, BIO *in_bio, X509 *pcert)
         break;
     case NID_pkcs7_enveloped:
         rsk = p7->d.enveloped->recipientinfo;
+IJON_SET((int)(data_body == NULL)); /* PATCHID:30783 */
         enc_alg = p7->d.enveloped->enc_data->algorithm;
         /* data_body is NULL if the optional EncryptedContent is missing. */
         data_body = p7->d.enveloped->enc_data->enc_data;
+IJON_CMP((unsigned long long)(data_body == NULL ? 1 : 0), (unsigned long long)1); /* PATCHID:26878 */
         OBJ_obj2txt(name, sizeof(name), enc_alg->algorithm, 0);
 
         (void)ERR_set_mark();
@@ -507,10 +525,14 @@ BIO *PKCS7_dataDecode(PKCS7 *p7, EVP_PKEY *pkey, BIO *in_bio, X509 *pcert)
 #ifdef MAGMA_ENABLE_FIXES
     if (data_body == NULL && in_bio == NULL) {
         ERR_raise(ERR_LIB_PKCS7, PKCS7_R_NO_CONTENT);
+IJON_CMP((unsigned long long)((data_body == NULL) && (in_bio == NULL)), (unsigned long long)1); /* PATCHID:29987 */
         goto err;
     }
 #endif
 #ifdef MAGMA_ENABLE_CANARIES
+
+IJON_CMP((unsigned long long)(in_bio == NULL ? 1 : 0), (unsigned long long)1); /* PATCHID:26238 */
+IJON_CMP((unsigned long long)(data_body == NULL ? 1 : 0), (unsigned long long)1); /* PATCHID:25839 */
         MAGMA_LOG("%MAGMA_BUG%", MAGMA_AND(data_body == NULL, in_bio == NULL));
 #endif
 
@@ -765,8 +787,10 @@ int PKCS7_dataFinal(PKCS7 *p7, BIO *bio)
 #else
 #ifdef MAGMA_ENABLE_CANARIES
     MAGMA_LOG("%MAGMA_BUG%", MAGMA_OR(p7 == NULL, p7->d.ptr == NULL));
+IJON_CMP((unsigned long long)p7, 0); /* PATCHID:15732 */
 #endif
 #endif
+IJON_CMP((unsigned long long)(p7 ? p7->d.ptr : 0), 0); /* PATCHID:15024 */
     ctx_tmp = EVP_MD_CTX_new();
     if (ctx_tmp == NULL) {
         ERR_raise(ERR_LIB_PKCS7, ERR_R_MALLOC_FAILURE);
@@ -774,11 +798,13 @@ int PKCS7_dataFinal(PKCS7 *p7, BIO *bio)
     }
 
     i = OBJ_obj2nid(p7->type);
+
     p7->state = PKCS7_S_HEADER;
 
     switch (i) {
     case NID_pkcs7_data:
         os = p7->d.data;
+IJON_CMP((unsigned long long)os, 0); /* PATCHID:14387 */
         break;
     case NID_pkcs7_signedAndEnveloped:
         /* XXXXXXXXXXXXXXXX */
@@ -796,6 +822,7 @@ int PKCS7_dataFinal(PKCS7 *p7, BIO *bio)
     case NID_pkcs7_enveloped:
         /* XXXXXXXXXXXXXXXX */
         os = p7->d.enveloped->enc_data->enc_data;
+IJON_CMP((unsigned long long)os, 0); /* PATCHID:14253 */
         if (os == NULL) {
             os = ASN1_OCTET_STRING_new();
             if (os == NULL) {
@@ -808,6 +835,7 @@ int PKCS7_dataFinal(PKCS7 *p7, BIO *bio)
     case NID_pkcs7_signed:
         si_sk = p7->d.sign->signer_info;
         os = PKCS7_get_octet_string(p7->d.sign->contents);
+IJON_CMP((unsigned long long)os, 0); /* PATCHID:14169 */
         /* If detached data then the content is excluded */
         if (PKCS7_type_is_data(p7->d.sign->contents) && p7->detached) {
             ASN1_OCTET_STRING_free(os);
@@ -818,6 +846,7 @@ int PKCS7_dataFinal(PKCS7 *p7, BIO *bio)
 
     case NID_pkcs7_digest:
         os = PKCS7_get_octet_string(p7->d.digest->contents);
+IJON_CMP((unsigned long long)os, 0); /* PATCHID:13603 */
         /* If detached data then the content is excluded */
         if (PKCS7_type_is_data(p7->d.digest->contents) && p7->detached) {
             ASN1_OCTET_STRING_free(os);
@@ -998,8 +1027,10 @@ int PKCS7_dataVerify(X509_STORE *cert_store, X509_STORE_CTX *ctx, BIO *bio,
 #else
 #ifdef MAGMA_ENABLE_CANARIES
     MAGMA_LOG("%MAGMA_BUG%", MAGMA_OR(p7 == NULL, p7->d.ptr));
+IJON_CTX(131); /* PATCHID:13347 */
 #endif
 #endif
+IJON_CMP((unsigned long long)(p7 ? p7->d.ptr : 0), 0); /* PATCHID:12324 */
     if (PKCS7_type_is_signed(p7)) {
         cert = p7->d.sign->cert;
     } else if (PKCS7_type_is_signedAndEnveloped(p7)) {
diff --git a/crypto/rsa/rsa_ameth.c b/crypto/rsa/rsa_ameth.c
index 46b3f1fce2..542e260d23 100644
--- a/crypto/rsa/rsa_ameth.c
+++ b/crypto/rsa/rsa_ameth.c
@@ -617,10 +617,14 @@ static int rsa_item_verify(EVP_MD_CTX *ctx, const ASN1_ITEM *it,
         ERR_raise(ERR_LIB_RSA, RSA_R_UNSUPPORTED_SIGNATURE_TYPE);
         return -1;
     }
+IJON_INC(1); /* PATCHID:19334 */
 #ifdef MAGMA_ENABLE_FIXES
     if (ossl_rsa_pss_to_ctx(ctx, NULL, sigalg, pkey) > 0) {
 #else
     int magma_int = ossl_rsa_pss_to_ctx(ctx, NULL, sigalg, pkey);
+IJON_CMP((unsigned long long)(magma_int + 1000), (unsigned long long)1000); /* PATCHID:18811 */
+IJON_MIN((unsigned long long)(magma_int + 1000)); /* PATCHID:18156 */
+IJON_SET((int)magma_int); /* PATCHID:17695 */
 #ifdef MAGMA_ENABLE_CANARIES
     MAGMA_LOG("%MAGMA_BUG%", magma_int < 0);
 #endif
diff --git a/crypto/x509/x509_obj.c b/crypto/x509/x509_obj.c
index c2f162d4bd..4ce44d4112 100644
--- a/crypto/x509/x509_obj.c
+++ b/crypto/x509/x509_obj.c
@@ -71,7 +71,10 @@ char *X509_NAME_oneline(const X509_NAME *a, char *buf, int len)
         l1 = strlen(s);
 
         type = ne->value->type;
+IJON_SET((int)type); /* PATCHID:50654 */
         num = ne->value->length;
+IJON_CMP((unsigned long long)num, 2048ULL); /* PATCHID:50638 */
+IJON_CMP((unsigned long long)num, 1024ULL); /* PATCHID:50059 */
         if (num > NAME_ONELINE_MAX) {
             ERR_raise(ERR_LIB_X509, X509_R_NAME_TOO_LONG);
             goto end;
@@ -83,6 +86,7 @@ char *X509_NAME_oneline(const X509_NAME *a, char *buf, int len)
             type == V_ASN1_PRINTABLESTRING ||
             type == V_ASN1_TELETEXSTRING ||
             type == V_ASN1_IA5STRING) {
+IJON_CTX(1ULL); /* PATCHID:49559 */
 #ifdef MAGMA_ENABLE_FIXES
             if (num > (int)sizeof(ebcdic_buf))
                 num = sizeof(ebcdic_buf);
@@ -100,6 +104,7 @@ char *X509_NAME_oneline(const X509_NAME *a, char *buf, int len)
 #endif
 #endif
             q = ebcdic_buf;
+IJON_CTX(0ULL); /* PATCHID:48619 */
         }
 #endif
 
diff --git a/crypto/x509/x509_req.c b/crypto/x509/x509_req.c
index 213f359221..8e78286fd7 100644
--- a/crypto/x509/x509_req.c
+++ b/crypto/x509/x509_req.c
@@ -44,12 +44,15 @@ X509_REQ *X509_to_X509_REQ(X509 *x, EVP_PKEY *pkey, const EVP_MD *md)
         goto err;
 
     pktmp = X509_get0_pubkey(x);
+IJON_MIN((unsigned long long)pktmp); /* PATCHID:52142 */
+IJON_CMP((unsigned long long)pktmp, 0); /* PATCHID:52066 */
 #ifdef MAGMA_ENABLE_FIXES
     if (pktmp == NULL)
         goto err;
 #endif
 #ifdef MAGMA_ENABLE_CANARIES
     MAGMA_LOG("%MAGMA_BUG%",pktmp == NULL);
+IJON_MAX((unsigned long long)(pktmp == NULL)); /* PATCHID:51145 */
 #endif
     i = X509_REQ_set_pubkey(ret, pktmp);
     if (!i)
diff --git a/crypto/x509/x509_vfy.c b/crypto/x509/x509_vfy.c
index afa2130d2b..47701e3d27 100644
--- a/crypto/x509/x509_vfy.c
+++ b/crypto/x509/x509_vfy.c
@@ -1068,7 +1068,10 @@ static int get_crl_sk(X509_STORE_CTX *ctx, X509_CRL **pcrl, X509_CRL **pdcrl,
     for (i = 0; i < sk_X509_CRL_num(crls); i++) {
         crl = sk_X509_CRL_value(crls, i);
         reasons = *preasons;
+IJON_SET((int)crl_score); /* PATCHID:11464 */
         crl_score = get_crl_score(ctx, &crl_issuer, &reasons, crl, x);
+IJON_DIST((long long)crl_score, (long long)best_score); /* PATCHID:10760 */
+IJON_CMP((unsigned long long)crl_score, (unsigned long long)best_score); /* PATCHID:9868 */
 #ifdef MAGMA_ENABLE_FIXES
         if (crl_score < best_score || crl_score == 0)
 #else
@@ -1080,8 +1083,10 @@ static int get_crl_sk(X509_STORE_CTX *ctx, X509_CRL **pcrl, X509_CRL **pdcrl,
         if (crl_score == best_score && best_crl != NULL) {
 #else
         if (crl_score == best_score) {
+IJON_INC((int)(best_crl == NULL)); /* PATCHID:9365 */
 #ifdef MAGMA_ENABLE_CANARIES
             MAGMA_LOG("%MAGMA_BUG%", best_crl == NULL);
+IJON_INC((int)(best_crl == NULL && crl_score == best_score)); /* PATCHID:8754 */
 #endif
 #endif
             int day, sec;
diff --git a/ssl/record/rec_layer_s3.c b/ssl/record/rec_layer_s3.c
index ebc4f9988c..5277c32339 100644
--- a/ssl/record/rec_layer_s3.c
+++ b/ssl/record/rec_layer_s3.c
@@ -1271,6 +1271,7 @@ int ssl3_read_bytes(SSL *s, int type, int *recvd_type, unsigned char *buf,
     SSL3_BUFFER *rbuf;
     void (*cb) (const SSL *ssl, int type2, int val) = NULL;
     int is_tls13 = SSL_IS_TLS13(s);
+IJON_CTX((unsigned long long)peek); /* PATCHID:8539 */
 
     rbuf = &s->rlayer.rbuf;
 
@@ -1368,6 +1369,8 @@ int ssl3_read_bytes(SSL *s, int type, int *recvd_type, unsigned char *buf,
         }
     } while (num_recs == 0);
     rr = &rr[curr_rec];
+IJON_INC((int)num_recs); /* PATCHID:8387 */
+IJON_SET((int)curr_rec); /* PATCHID:7568 */
 
     if (s->rlayer.handshake_fragment_len > 0
             && SSL3_RECORD_get_type(rr) != SSL3_RT_HANDSHAKE
@@ -1446,11 +1449,13 @@ int ssl3_read_bytes(SSL *s, int type, int *recvd_type, unsigned char *buf,
         }
 
         totalbytes = 0;
+IJON_SET((int)totalbytes); /* PATCHID:7334 */
         do {
             if (len - totalbytes > SSL3_RECORD_get_length(rr))
                 n = SSL3_RECORD_get_length(rr);
             else
                 n = len - totalbytes;
+IJON_MIN((unsigned long long)n); /* PATCHID:6584 */
 
             memcpy(buf, &(rr->data[rr->off]), n);
             buf += n;
@@ -1462,6 +1467,7 @@ int ssl3_read_bytes(SSL *s, int type, int *recvd_type, unsigned char *buf,
             } else {
 #else
 #ifdef MAGMA_ENABLE_CANARIES
+IJON_CMP((unsigned long long)peek, 1); /* PATCHID:6409 */
             MAGMA_LOG("%MAGMA_BUG%", MAGMA_AND(peek, \
                                 MAGMA_AND(SSL3_RECORD_get_length(rr) == 0, \
                                           !SSL3_RECORD_is_read(rr))));
diff --git a/ssl/statem/statem.c b/ssl/statem/statem.c
index 9e79b27131..422c7cd396 100644
--- a/ssl/statem/statem.c
+++ b/ssl/statem/statem.c
@@ -505,6 +505,7 @@ static void init_read_state_machine(SSL *s)
 static int grow_init_buf(SSL *s, size_t size) {
 
     size_t msg_offset = (char *)s->init_msg - s->init_buf->data;
+IJON_SET((int)msg_offset); /* PATCHID:23803 */
 
     if (!BUF_MEM_grow_clean(s->init_buf, (int)size))
         return 0;
@@ -516,6 +517,7 @@ static int grow_init_buf(SSL *s, size_t size) {
     s->init_msg = s->init_buf->data + msg_offset;
 #endif
 #ifdef MAGMA_ENABLE_CANARIES
+IJON_CMP((unsigned long long)s->init_msg, (unsigned long long)(s->init_buf->data + msg_offset)); /* PATCHID:23087 */
     MAGMA_LOG("%MAGMA_BUG%", s->init_msg != (s->init_buf->data + msg_offset));
 #endif
 
diff --git a/ssl/statem/statem_clnt.c b/ssl/statem/statem_clnt.c
index 98b020a27f..7423d5c964 100644
--- a/ssl/statem/statem_clnt.c
+++ b/ssl/statem/statem_clnt.c
@@ -2908,12 +2908,15 @@ static int tls_construct_cke_dhe(SSL *s, WPACKET *pkt)
     int ret = 0;
 
     skey = s->s3.peer_tmp;
+IJON_SET((int)(skey != NULL)); /* PATCHID:4811 */
     if (skey == NULL) {
         SSLfatal(s, SSL_AD_INTERNAL_ERROR, ERR_R_INTERNAL_ERROR);
         goto err;
     }
 
     ckey = ssl_generate_pkey(s, skey);
+IJON_SET((int)(ckey == NULL)); /* PATCHID:4481 */
+IJON_CMP((unsigned long long)ckey, 0); /* PATCHID:4327 */
 #ifdef MAGMA_ENABLE_FIXES
     if (ckey == NULL) {
         SSLfatal(s, SSL_AD_INTERNAL_ERROR, ERR_R_INTERNAL_ERROR);
@@ -2922,6 +2925,7 @@ static int tls_construct_cke_dhe(SSL *s, WPACKET *pkt)
 #endif
 #ifdef MAGMA_ENABLE_CANARIES
     MAGMA_LOG("%MAGMA_BUG%",ckey == NULL);
+IJON_INC((int)(ckey == NULL)); /* PATCHID:3694 */
 #endif
 
     if (ssl_derive(s, ckey, skey, 0) == 0) {
@@ -2988,17 +2992,21 @@ static int tls_construct_cke_ecdhe(SSL *s, WPACKET *pkt)
 #endif
 #ifdef MAGMA_ENABLE_CANARIES
     MAGMA_LOG("%MAGMA_BUG%",ckey == NULL);
+IJON_SET((int)(ckey == NULL)); /* PATCHID:2838 */
 #endif
 
     if (ssl_derive(s, ckey, skey, 0) == 0) {
         /* SSLfatal() already called */
         goto err;
+
     }
 
     /* Generate encoding of client key */
     encoded_pt_len = EVP_PKEY_get1_encoded_public_key(ckey, &encodedPoint);
+IJON_SET((int)encoded_pt_len); /* PATCHID:1477 */
 
     if (encoded_pt_len == 0) {
+IJON_INC((int)(encoded_pt_len == 0)); /* PATCHID:1353 */
         SSLfatal(s, SSL_AD_INTERNAL_ERROR, ERR_R_EC_LIB);
         goto err;
     }
@@ -3009,6 +3017,7 @@ static int tls_construct_cke_ecdhe(SSL *s, WPACKET *pkt)
     }
 
     ret = 1;
+IJON_SET((int)ret); /* PATCHID:880 */
  err:
     OPENSSL_free(encodedPoint);
     EVP_PKEY_free(ckey);
diff --git a/ssl/statem/statem_dtls.c b/ssl/statem/statem_dtls.c
index bec4daf3a1..4d9dc57bf9 100644
--- a/ssl/statem/statem_dtls.c
+++ b/ssl/statem/statem_dtls.c
@@ -443,6 +443,8 @@ static int dtls1_preprocess_fragment(SSL *s, struct hm_header_st *msg_hdr)
     msg_len = msg_hdr->msg_len;
     frag_off = msg_hdr->frag_off;
     frag_len = msg_hdr->frag_len;
+IJON_CMP((unsigned long long)(frag_off + frag_len), (unsigned long long)msg_len); /* PATCHID:25254 */
+IJON_MAX((unsigned long long)msg_len); /* PATCHID:24784 */
 
 #ifdef MAGMA_ENABLE_FIXES
     /* sanity checking */
@@ -459,6 +461,7 @@ static int dtls1_preprocess_fragment(SSL *s, struct hm_header_st *msg_hdr)
 #endif
 
     if (s->d1->r_msg_hdr.frag_off == 0) { /* first fragment */
+IJON_CTX(300); /* PATCHID:24569 */
         /*
          * msg_len is limited to 2^24, but is effectively checked against
          * dtls_max_handshake_message_len(s) above
@@ -474,6 +477,7 @@ static int dtls1_preprocess_fragment(SSL *s, struct hm_header_st *msg_hdr)
         s->d1->r_msg_hdr.type = msg_hdr->type;
         s->d1->r_msg_hdr.seq = msg_hdr->seq;
     } else if (msg_len != s->d1->r_msg_hdr.msg_len) {
+IJON_CTX(247); /* PATCHID:24247 */
         /*
          * They must be playing with us! BTW, failure to enforce upper limit
          * would open possibility for buffer overrun.
diff --git a/ssl/t1_lib.c b/ssl/t1_lib.c
index be85b39349..35f8ab70b4 100644
--- a/ssl/t1_lib.c
+++ b/ssl/t1_lib.c
@@ -1886,11 +1886,13 @@ SSL_TICKET_STATUS tls_decrypt_ticket(SSL *s, const unsigned char *etick,
      * checks on ticket.
      */
     mlen = ssl_hmac_size(hctx);
+IJON_SET((int)mlen); /* PATCHID:5808 */
     if (mlen == 0) {
         ret = SSL_TICKET_FATAL_ERR_OTHER;
         goto end;
     }
 
+IJON_CMP((unsigned long long)eticklen, 48ULL); /* PATCHID:5734 */
     /* Sanity check ticket length: must exceed keyname + IV + HMAC */
 #ifdef MAGMA_ENABLE_FIXES
     if (eticklen <=
