diff --git a/ldebug.c b/ldebug.c
index ecb3536d..5ee25217 100644
--- a/ldebug.c
+++ b/ldebug.c
@@ -13,6 +13,7 @@
 #include <stdarg.h>
 #include <stddef.h>
 #include <string.h>
+#include <stdio.h>

 #include "lua.h"

@@ -188,13 +188,19 @@ static const char *upvalname (const Proto *p, int uv) {
 static const char *findvararg (CallInfo *ci, int n, StkId *pos) {
   if (clLvalue(s2v(ci->func))->p->is_vararg) {
     int nextra = ci->u.l.nextraargs;
+IJON_SET(nextra); /* PATCHID:10561 */
 #ifdef MAGMA_ENABLE_FIXES
     if (n >= -nextra) {  /* 'n' is negative */
       *pos = ci->func - nextra - (n + 1);
 #else
     if (n <= nextra) {
+IJON_CTX(471); /* PATCHID:10081 */
 #ifdef MAGMA_ENABLE_CANARIES
+IJON_DIST((long long)n, (long long)(INT_MAX - nextra + 1)); /* PATCHID:9845 */
+IJON_SET(n); /* PATCHID:9713 */
       MAGMA_LOG("LUA001", INT_MAX - nextra <= (n - 1));
+IJON_MAX((unsigned long long)(n - 1)); /* PATCHID:8756 */
+IJON_CMP((unsigned long long)(n - 1), (unsigned long long)(INT_MAX - nextra)); /* PATCHID:7815 */
 #endif
       *pos = ci->func - nextra + (n - 1);
 #endif
@@ -846,13 +852,16 @@ static int changedline (const Proto *p, int oldpc, int newpc) {
 #else
 #ifdef MAGMA_ENABLE_CANARIES
   MAGMA_LOG("LUA002", p->lineinfo == NULL);
+IJON_CMP((unsigned long long)p->lineinfo, (unsigned long long)0); /* PATCHID:5511 */
 #endif
 #endif
   if (newpc - oldpc < MAXIWTHABS / 2) {  /* not too far apart? */
+IJON_DIST((long long)(newpc - oldpc), (long long)0); /* PATCHID:4824 */
     int delta = 0;  /* line diference */
     int pc = oldpc;
     for (;;) {
       int lineinfo = p->lineinfo[++pc];
+IJON_MAX((unsigned long long)pc); /* PATCHID:4517 */
       if (lineinfo == ABSLINEINFO)
         break;  /* cannot compute delta; fall through */
       delta += lineinfo;
@@ -905,18 +914,24 @@ int luaG_traceexec (lua_State *L, const Instruction *pc) {
   if (counthook)
     luaD_hook(L, LUA_HOOKCOUNT, -1, 0, 0);  /* call count hook */
   if (mask & LUA_MASKLINE) {
+IJON_SET((int)(mask & LUA_MASKLINE)); /* PATCHID:3573 */
 #ifdef MAGMA_ENABLE_FIXES
     /* 'L->oldpc' may be invalid; use zero in this case */
     int oldpc = (L->oldpc < p->sizecode) ? L->oldpc : 0;
 #else
     const Proto *p = ci_func(ci)->p;
+IJON_SET((int)p->sizecode); /* PATCHID:3449 */
+IJON_SET((int)L->oldpc); /* PATCHID:3362 */
 #endif
     int npci = pcRel(pc, p);
+IJON_SET((int)npci); /* PATCHID:2526 */
 #ifdef MAGMA_ENABLE_FIXES
     if (npci <= oldpc ||  /* call hook when jump back (loop), */
         changedline(p, oldpc, npci)) {  /* or when enter new line */
 #else
 #ifdef MAGMA_ENABLE_CANARIES
+IJON_CTX(438); /* PATCHID:1849 */
+IJON_CMP((unsigned long long)L->oldpc, (unsigned long long)p->sizecode); /* PATCHID:979 */
     MAGMA_LOG("LUA004", pcRel(L->oldpc, p) < 0);
 #endif
     if (npci == 0 ||
diff --git a/liolib.c b/liolib.c
index 12dfb7b7..eeb457a3 100644
--- a/liolib.c
+++ b/liolib.c
@@ -16,6 +16,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <stdio.h>

 #include "lua.h"

@@ -290,11 +290,16 @@ static int io_pclose (lua_State *L) {
 static int io_popen (lua_State *L) {
   const char *filename = luaL_checkstring(L, 1);
   const char *mode = luaL_optstring(L, 2, "r");
+IJON_STRDIST(mode, "rw"); /* PATCHID:7571 */
+IJON_STRDIST(mode, "invalid"); /* PATCHID:7321 */
+IJON_STRDIST(mode, "x"); /* PATCHID:7057 */
+IJON_SET((int)mode[0]); /* PATCHID:6386 */
   LStream *p = newprefile(L);
 #ifdef MAGMA_ENABLE_FIXES
   luaL_argcheck(L, l_checkmodep(mode), 2, "invalid mode");
 #else
 #ifdef MAGMA_ENABLE_CANARIES
+IJON_INC(1); /* PATCHID:5738 */
   MAGMA_LOG("LUA003", !l_checkmodep(mode));
 #endif
 #endif
