diff --git a/src/where.c b/src/where.c
index 18f3b6d..e8bba4d 100644
--- a/src/where.c
+++ b/src/where.c
@@ -2831,6 +2831,7 @@ static int whereLoopAddBtreeIndex(
   ** On the other hand, the extra seeks could end up being significantly
   ** more expensive.  */
   assert( 42==sqlite3LogEst(18) );
+
   if( saved_nEq==saved_nSkip
    && saved_nEq+1<pProbe->nKeyCol
    && saved_nEq==pNew->nLTerm
@@ -2842,6 +2843,7 @@ static int whereLoopAddBtreeIndex(
    && (rc = whereLoopResize(db, pNew, pNew->nLTerm+1))==SQLITE_OK
 #endif
   ){
+IJON_MAX((int)pNew->nLTerm); /* PATCHID:18814 */
 #ifdef MAGMA_ENABLE_CANARIES
       MAGMA_LOG("SQL013", pNew->nLSlot < (pNew->nLTerm+1));
 #endif
